ee/hogai/graph/mixins.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+from abc import ABC, abstractmethod
+from uuid import UUID
+
+from django.utils import timezone
+
+from ee.models import Conversation, CoreMemory
+from posthog.models import Team
+from posthog.models.user import User
+
+
+class AssistantContextMixin(ABC):
+    @property
+    @abstractmethod
+    def _team(self) -> Team: ...
+
+    @property
+    @abstractmethod
+    def _user(self) -> User: ...
```

Comment:
Why not just define the attributes plainly like so:

```suggestion
    _team: Team
    _user: User
```

Reply:
It conflicts with the MaxTool class being a Pydantic model:
- Internally, Team and User are lazily initialized, so the type hint must be `Optional[...]`.
- Although this can be changed to become constructor parameters, mypy will treat them as optional anyway (Pydantic behavior). This means that there won't be static type checking for MaxTools, but rather a runtime one, which feels wrong to me (but maybe it's not?).

Example:
```python
# mypy is green, but it raises a runtime validation exception.
tool = MaxTool()
```